src/App.jsx

```diff
@@ -14,45 +15,107 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const ALPHABET = 'by alphabet';
+const LENGTH = 'by length';
+
+function getPreparedGoods(goods, { methodSort, reverse }) {
+  const copyOfGoods = [...goods];
+
+  if (methodSort) {
+    copyOfGoods.sort((good1, good2) => {
+      switch (methodSort) {
+        case ALPHABET:
+          return good1.localeCompare(good2);
+        case LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    return copyOfGoods.reverse();
+  }
+
+  return copyOfGoods;
+}
+
+export const App = () => {
+  const [methodSort, setMethodSort] = useState('');
+  const [reset, setReset] = useState(false);
+  const [reverse, setReverse] = useState(false);
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const visibleGoods = getPreparedGoods(goodsFromServer,
+    { methodSort, reverse });
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={methodSort === ALPHABET
+            ? 'button is-info'
+            : 'button is-info is-light'}
+          onClick={() => {
+            setMethodSort(ALPHABET);
+            setReset(true);
+          }}
+        >
+          Sort alphabetically
+        </button>
+
+        <button
+          type="button"
+          className={methodSort === LENGTH
+            ? 'button is-success'
+            : 'button is-success is-light'}
+          onClick={() => {
+            setMethodSort(LENGTH);
+            setReset(true);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={reverse
+            ? 'button is-warning'
+            : 'button is-warning is-light'}
+          onClick={() => {
+            setReset(true);
+            setReverse(!reverse);
+          }}
+        >
+          Reverse
+        </button>
+
+        {reset && (
```

Comment:
```suggestion
        {methodSort || reverse && (
```

and you can remove reverse state